### PR TITLE
Add Overview of Benchmark and Model Zoo

### PR DESCRIPTION
**Pull request by Keiku**
## Motivation

According to issue #6868, I created a table that makes the list of model zoo easy to understand. I hope this makes it easier for everyone to understand the models that mmdetection supports. For similar documentation, I referred to the [Overview of Kit Structures in Paddle Detection](https://github.com/PaddlePaddle/PaddleDetection#overview-of-kit-structures).  **At a minimum, separating Object Detection and Instance Segmentation makes it easier for users to understand the models supported by mmdetection.**

## Modification

The list of Benchmark and model zoo in `README.md` is changed to table.

## BC-breaking (Optional)

The Chinese version of the document can also be changed. We should also change the list of model zoos in the link (`docs/en/model_zoo.md`).

## Use cases (Optional)

Only the models that exist in config are listed. Not all features of mmdetection are listed.

## Checklist

1. I left a comment `# Not officially listed` as a list that hasn't been officially listed yet. Please check if it matches.
2. Check if the table division is correct.
3. Check if the listed items correspond to the table category.
4. The order of the list is the submission date of the paper and the year of adoption at the conference, but please make sure that they match. Reference information is provided [here](https://github.com/Keiku/mmdet-overview-of-kit-structures).
5. This table is created based on `v2.19.1`, so please add the update.

### CODE REVIEW

**Comment by RangiLyu** [comment]
Thanks for your contribution!

How about switching to a format like this?


<div align="center">
 <b>Architectures</b>
</div>

Object Detection | Instance Segmentation | Panoptic Segmentation | Distillation
-- | -- | -- | --
Fast R-CNN (ICCV'2015) Faster R-CNN (NeurIPS'2015) SSD (ECCV'2016) RetinaNet (ICCV'2017) Cascade R-CNN (CVPR'2018) YOLOv3 (ArXiv'2018) CornerNet (ECCV'2018) Grid R-CNN (CVPR'2019) CenterNet (CVPR'2019) Libra R-CNN (CVPR'2019) | Mask R-CNN (ICCV'2017) Cascade Mask R-CNN (CVPR'2018)  Mask Scoring R-CNN (CVPR'2019) Hybrid Task Cascade (CVPR'2019) YOLACT (ICCV'2019) InstaBoost (ICCV'2019) SOLO (ECCV'2020) | Panoptic FPN | Localization Distillation (ArXiv'2021) Label Assignment Distillation (WACV'2022)

<div align="center">
 <b>Components</b>
</div>

Backbones | Necks | Heads | Loss
-- | -- | -- | --
VGG (ICLR'2015) ResNet (CVPR'2016) ResNeXt (CVPR'2017) MobileNetV2 (CVPR'2018) HRNet (CVPR'2019) Generalized Attention (ICCV'2019) GCNet (ICCVW'2019) Res2Net (TPAMI'2020) RegNet (CVPR'2020) ResNeSt (ArXiv'2020) PVT (ICCV'2021) Swin (CVPR'2021) PVTv2 (ArXiv'2021) |   |   |

**Comment by Keiku** [comment]
@RangiLyu Thank you for your instructions. I refactored the table at your instructions.
@hhaAndroid @ZwwWayne Is there anything you would like to change?

**Comment by RangiLyu** [comment]
image

Looks great! And I have some suggestions:

- Rename "Architectures"  to "Algorithms"

- The comments like `# Not officially listed` etc. can be deleted.

- And about the component section, we can remove the `Heads` and move those papers to the "Algorithms" table.

**Comment by RangiLyu** [comment]
@ZwwWayne @hhaAndroid Any suggestions?

**Comment by ZwwWayne** [comment]
LGTM. Can be merged after cleaning '# Not officially listed'

**Comment by Keiku** [comment]
@RangiLyu Updated for reviews.

The Algorithms are as follows.

image

It seems that the centering of Components has failed, but I'm not sure.

image

@RangiLyu When I cover mmdetection's configs, it looks like a list like the link below. Some items (configs) are missing in the above table, is that okay?

https://github.com/Keiku/mmdet-overview-of-kit-structures

File: README.md [hunk 1]
```diff
@@ -69,87 +69,158 @@ Please refer to [changelog.md](docs/en/changelog.md) for details and release his
...
-- [x] [YOLOX (ArXiv'2021)](configs/yolox/README.md)
-- [x] [SOLO (ECCV'2020)](configs/solo/README.md)
-- [x] [QueryInst (ICCV'2021)](configs/queryinst/README.md)
-- [x] [TOOD (ICCV'2021)](configs/tood/README.md)
-</details>
+<div style="text-align: center;">
+  <td>
+    <b>Algorithms</b>
+  </td>
+</div>
```
Comment (RangiLyu) [inline README.md:80]:
```suggestion
<div align="center">
 <b>Architectures</b>
</div>
```

File: README.md [hunk 2]
```diff
@@ -69,87 +69,158 @@ Please refer to [changelog.md](docs/en/changelog.md) for details and release his
...
+</td>
+    </tr>
+  </tbody>
+</table>
+
+<div style="text-align: center;">
+  <td>
+    <b>Components</b>
+  </td>
+</div>
```
Comment (RangiLyu) [inline README.md:171]:
```suggestion
<div align="center">
 <b>Components</b>
</div>
```

**Comment by RangiLyu** [comment]
> When I cover mmdetection's configs, it looks like a list like the link below. Some items (configs) are missing in the above table, is that okay?

It's okay.

**Comment by Keiku** [comment]
@RangiLyu Excuse me. Let me ask you a question.

* What is the difference between Faster R-CNN (NeurIPS'2015) and RPN (NeurIPS'2015)? Is there any problem with RPN in the Object Detection category?

What about the following categories?

* "Distillation" may be set to "Other", and "Distillation" and "Contrastive Learning" may be included in "Other".

* It may be possible to put "Common" ((OHEM (CVPR'2016), Group Normalization (ECCV'2018), DCNv2 (CVPR'2019), Weight Standardization (ArXiv'2019), Prime Sample Attention (CVPR'2020), Large-Scale Jitter)) in Components.

There was also a new README.md PR recently. Probably conflict.

File: README.md (hunk 1 above)
Comment (Keiku) [reply README.md]: Thank you for your instructions. I fixed it.

File: README.md (hunk 2 above)
Comment (Keiku) [reply README.md]: Thank you for your instructions. I fixed it.

**Comment by RangiLyu** [comment]
> * What is the difference between Faster R-CNN (NeurIPS'2015) and RPN (NeurIPS'2015)? Is there any problem with RPN in the Object Detection category?

RPN is the first stage in Faster R-CNN, It's Okay to put it in Object Detection category.


> * "Distillation" may be set to "Other", and "Distillation" and "Contrastive Learning" may be included in "Other".
> * It may be possible to put "Common" ((OHEM (CVPR'2016), Group Normalization (ECCV'2018), DCNv2 (CVPR'2019), Weight Standardization (ArXiv'2019), Prime Sample Attention (CVPR'2020), Large-Scale Jitter)) in Components.

Yes, you can add "Other" both to Algorithms and Components.

**Comment by RangiLyu** [comment]
@Keiku Hi, any update? We hope this PR can be merged in the coming version of MMDetection.

**Comment by Keiku** [comment]
@RangiLyu I'll work on it immediately.

**Comment by Keiku** [comment]
@RangiLyu Please check for updates.

* Since the original material of Strong Baselines is Copy-Paste paper, I chose CVPR'2021. Please refer to the following link. [New, improved Detectron2 Mask R-CNN baselines](https://ai.facebook.com/blog/advancing-computer-vision-research-with-new-detectron2-mask-r-cnn-baselines/)

**Comment by ZwwWayne** [comment]
LGTM. Thank you for the kind modification @Keiku !

**Review by RangiLyu — APPROVED** [review]
LGTM

**Review by ZwwWayne — APPROVED** [review]

**Comment by RangiLyu** [comment]
This PR resolves #7030

**Pull request merged**